R/estimateDiversity.R

```diff
@@ -411,6 +440,39 @@ setMethod("estimateFaith", signature = c(x="TreeSummarizedExperiment", tree="mis
     return(faiths)
 }
 
+.calc_log_modulo_skewness <- function(mat, quantile = 0.5, num_of_classes = 50, ...){
+    # quantile must be a numeric value between 0-1
+    if( !( is.numeric(quantile) && (quantile >= 0 && quantile <= 1) ) ){
+        stop("'quantile' must be a numeric value between 0-1.",
+             call. = FALSE)
+    }
+    # num_of_classes must be a positive numeric value
+    if( !( is.numeric(num_of_classes) && num_of_classes > 0 ) ){
+        stop("'num_of_classes' must be a positive numeric value.",
+             call. = FALSE)
+    }
+    # Determine the quantile point.
+    quantile_point <- quantile(max(mat), quantile)
+    
+    # Tabulate the arithmetic abundance classes. Use the same classes
+    # for all samples for consistency    
+    cutpoints <- c(seq(0, quantile_point, length=num_of_classes), Inf)
+    
+    # Check skewness of the abundance classes for each sample
+    r <- apply(mat, 2, function(x) {
+        .calc_skewness(table(cut(x, cutpoints)))
+    })
```

Comment:
Is the result of `table` always of the same length? If yes, I think you could construct a matrix and vectorize `.calc_skewness`

Reply:
Vectorized